### DISCUSSION TITLE
how to send request with header

### DISCUSSION THREAD

**Discussion by shahensaifullah** [nuxt/nuxt#17164] [Questions]
In axios, usually i use onRequest to set a value to the header method in plugins. When Axios send a request, it automatically sends a request with that header. i don't need to write duplicate code repeatedly. 
but in nuxt 3 cannot use axios 😪.
So how can I create an instance for $fetch method with the base URL and use that $fetch without repeating the same code?

**Comment by mihail727** [comment]
maybe it help you
https://github.com/nuxt/framework/discussions/4597

**Reply by shahensaifullah** [reply]
hey i want to cookies are not sending to the server and I want to set a default csrf token to every request.
```
    const accessCookie = useCookie('csrf_bongo') === undefined ? null : useCookie('csrf_bongo');
    options.headers['X-CSRFToken'] = accessCookie
```
but this is not set to header, shows undefined data cannot set to header. what should I do for that?
